Makefile

```diff
@@ -87,6 +87,9 @@ install: bdist lint ## Build distribution and install
 test: lint ## Run unit tests
 	pytest -v elyra
 
+test-ui: lint ## Run frontend tests
```

Comment:
A two comments here:

1. Why isn't this part of `make test`?

2. Maybe move `npm test` to make test and have this run `npm run test-debug` instead

Reply:
1. I think the idea behind having a separate test is that you don't necessarily always want to run these tests at the same time.
2. @lresende What do you think?

Reply:
I think there should be a way to run the tests separately.  I would recommend the following:

Create separate lint and test tags.  

```Makefile
lint-ui:  # runs all the stuff necessary for dealing with linting on the front end

lint-server: # runs all the stuff necessary for dealing with linting on the server

lint: lint-server lint-ui

test-ui: lint-ui # runs all the stuff necessary for testing the front end

test-server: lint-server  # runs all the stuff necessary for testing the server
    pytest -v elyra

test: lint test-server test-ui
```
The CI scripts will call `make test` just as they do today - as will the `install` target.  So there should be no changes required for CI - unless test-ui needs some additional support.

Reply:
I like @kevin-bates suggestion, it gives us flexibility with testing, but I still think we should have a make target for the frontend `test-debug` as well

Reply:
here's git diff for implementing @kevin-bates suggested changes:
```
diff --git a/Makefile b/Makefile
index bdb5d10..075149a 100644
--- a/Makefile
+++ b/Makefile
@@ -49,10 +49,14 @@ yarn-install:
 test-dependencies:
        @pip install -q -r test_requirements.txt
 
-lint: test-dependencies ## Run linters
+lint-server: test-dependencies
+       flake8 elyra
+
+lint-ui:
        yarn run prettier
        yarn run eslint
-       flake8 elyra
+
+lint: lint-ui lint-server ## Run linters
 
 lerna-build: yarn-install
        export PATH=$$(pwd)/node_modules/.bin:$$PATH && lerna run build
@@ -84,12 +88,17 @@ install: bdist lint ## Build distribution and install
        jupyter serverextension list
        jupyter labextension list
 
-test: lint ## Run unit tests
+test-server: lint-server ## Run unit tests
        pytest -v elyra
 
-test-ui: lint ## Run frontend tests
+test-ui: lint-ui ## Run frontend tests
        npm test
 
+test-ui-debug: lint-ui ## Run frontend tests
+       npm run test-debug
+
+test: test-server test-ui ## Run all tests
+
 install-backend: ## Build and install backend
        python setup.py bdist_wheel --dev
        pip install --upgrade dist/elyra-*-py3-none-any.whl
```

Reply:
note: to "apply" that git-diff copy the block above into a file and run `git apply <filename>`